Lib/pathlib.py

```diff
@@ -407,6 +271,51 @@ def __reduce__(self):
         # when pickling related paths.
         return (self.__class__, tuple(self._parts))
 
+    @classmethod
+    def _split_root(cls, part):
```

Comment:
I plan to tidy this method up and move its implementation into `ntpath` and `posixpath`, but I'd like to do it in a subsequent PR that allows users to swap in their own path normalization logic.

Reply:
Is calling `normpath()` and `splitdrive()` on the anchor just to support a "//" root on POSIX? I agree that this should use a new `splitroot()` function, implemented separately in `ntpath` and `posixpath`.

Reply:
It's used to normalize roots on both POSIX (like "///etc" -> "/etc") and Windows (like "C:\\foo" -> "C:\foo"). This could be implemented more efficiently (and obviously) with dedicated functions in `ntpath` and `posixpath`. I think I'll have a PR for that within the next six months.